test/proto/BUILD

```diff
@@ -60,3 +66,24 @@ scalapb_proto_library(
     with_grpc = True,
     deps = [":test_service"],
 )
+
+scala_proto_srcjar(
+    name = "test1_proto_scala",
+    deps = ["//test/proto2:test"],
+    generator = "@io_bazel_rules_scala//src/scala/scripts:scalapb_generator")
+
+scala_proto_srcjar(
+    name = "test2_proto_scala",
+    deps = [":test2"],
+    blacklisted_protos = ["//test/proto2:test"],
```

Comment:
this feels like the wrong approach to me. The thrift rules don't work this way. It seems wrong to me that we are transitively building all the protos in each target. Instead, I think each target should only build what they depend on (as the thrift rules do).

If we do this, there is no need for blacklisting, since we only ever build one target.

In the thrift rules, we used aspects to achieve this:
https://github.com/bazelbuild/rules_scala/blob/master/twitter_scrooge/twitter_scrooge.bzl#L404

Reply:
@johnynek thanks for the reply. 

I just want to make sure I understand you correctly. Let's say we have targets like this
```
proto_library(
    name = "a_proto",
    srcs = ["a.proto"]
)

proto_library(
    name = "b_proto",
    srcs = ["b.proto"],
    deps = [":a_proto"]
)
```

and if we have following target
```
scalapb_proto_library(
    name = "scala_proto",
    deps = [":b_proto"],
)
```

scala code should be generated only for `b.proto`?